src/wasm/wasm-emscripten.cpp

```diff
@@ -588,8 +592,11 @@ struct AsmConstWalker : public LinearExecutionWalker<AsmConstWalker> {
   Index resolveConstIndex(Expression* arg,
                           std::function<void(Expression*)> reportError);
   Const* resolveConstAddr(Expression* arg, const Name& target);
+  Name* resolveGlobalGet(Expression* arg, const Name& target);
```

Comment:
is there a reason this can't be a `Name&`? that would be better I think.

Reply:
(but I don't fully understand why we need to change the value - I defer to the people that know this code better)

Reply:
We need to change the value because the `global.get` could have come from a string of `local.get`/`local.set`s. We have to replace it at the source, or we'll end up with an undefined symbol.

Reply:
Reverted this back to `Name*` because it might fail to get anything. In that case, we need to return `nullptr` so the caller can skip it.